### PR TITLE
1.0.0-0alpha7

### PR DESCRIPTION
**Pull request by 3Hren**
Changes:
------------
- Add TCP sink.
- Add process and thread fields in JSON layout.
- Flush interval for file sink.
- More documentation.
- Extend config interface allowing to check node types (breaks vtable size).
- Remove `filter` method from sinks (breaks vtables).

**All code that uses sinks and configs directly is required to be recompiled.**

### CODE REVIEW

**Comment by 3Hren** [comment]
@antmat TAL if you want.

File: include/blackhole/detail/sink/socket/tcp.hpp [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+#include "blackhole/cpp17/string_view.hpp"
+#include "blackhole/extensions/format.hpp"
+#include "blackhole/sink.hpp"
+#include "blackhole/sink/socket/tcp.hpp"
+
+namespace blackhole {
+inline namespace v1 {
+namespace sink {
+namespace socket {
+namespace {
```
Comment (antmat) [inline include/blackhole/detail/sink/socket/tcp.hpp]: Looks strange

File: include/blackhole/detail/sink/socket/tcp.hpp [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+        boost::asio::io_service io_service;
+        std::unique_ptr<socket_type> socket;
+
+        mutable std::mutex mutex;
+    };
+
+    std::unique_ptr<data_t> data;
+
+public:
+    tcp_t(std::string host, std::uint16_t port) :
```
Comment (antmat) [inline include/blackhole/detail/sink/socket/tcp.hpp]: "inline" here and later?

Comment (3Hren) [inline include/blackhole/detail/sink/socket/tcp.hpp]: Nope.

**Pull request merged**